bleach/html5lib_shim.py

```diff
@@ -557,9 +557,11 @@ def match_entity(stream):
     # Handle character entities
     while stream and stream[0] not in end_characters:
         c = stream.pop(0)
-        if not ENTITIES_TRIE.has_keys_with_prefix(possible_entity):
-            break
         possible_entity += c
+        if not ENTITIES_TRIE.has_keys_with_prefix(possible_entity):
+            # If it's not a prefix, then it's not an entity and we're
+            # out
+            return None
```

Comment:
This block mirrors the structure of the one that handles numeric entities, but it was awkwardly wrong and had been since I wrote it in 2017:

https://github.com/mozilla/bleach/commit/d8f7d1d651a6b5a4c8e5a5932ea4fd61769a60d4

:(